include/lsst/afw/detection/Footprint.h

```diff
@@ -340,212 +235,19 @@ class Footprint : public lsst::daf::base::Citizen,
- */
-std::vector<lsst::afw::geom::Box2I> footprintToBBoxList(Footprint const& foot);
-
-/**
- * \brief Set all image pixels in a Footprint to a given value
- *
- * \return value
- */
-template<typename ImageT>
-typename ImageT::Pixel setImageFromFootprint(ImageT *image,
-                                             Footprint const& footprint,
-                                             typename ImageT::Pixel const value);
-
-/**
- * \brief Set all image pixels in a set of Footprint%s to a given value
- *
- * \return value
- */
-template<typename ImageT>
-typename ImageT::Pixel setImageFromFootprintList(ImageT *image,
-                                                 CONST_PTR(std::vector<PTR(Footprint)>) footprints,
-                                                 typename ImageT::Pixel  const value);
-
-/**
- * \brief Set all image pixels in a set of Footprint%s to a given value
- *
- * \return value
- */
-template<typename ImageT>
-typename ImageT::Pixel setImageFromFootprintList(ImageT *image,
-                                                 std::vector<PTR(Footprint)> const& footprints,
-                                                 typename ImageT::Pixel  const value);
-
-/**
- * \brief OR bitmask into all the Mask's pixels that are in the Footprint
- *
- * \return bitmask
- */
-template<typename MaskT>
-MaskT setMaskFromFootprint(lsst::afw::image::Mask<MaskT> *mask,
-                           Footprint const& footprint,
-                           MaskT const bitmask);
-
-/**
- * \brief (AND ~bitmask) all the Mask's pixels that are in the
- * Footprint; that is, set to zero in the Mask-intersecting-Footprint
- * all bits that are 1 in then bitmask.
- *
- * \return bitmask
- */
-template<typename MaskT>
-MaskT clearMaskFromFootprint(lsst::afw::image::Mask<MaskT> *mask,
-                             Footprint const& footprint,
-                             MaskT const bitmask);
-
-/**
- Copies pixels from input image to output image within the Footprint's
- area.
-
- The input and output image must be the same type -- either Image or
- MaskedImage.
-
- Respects image xy0, so the footprints should be defined in the
- "PARENT" frame.
-
- Pixels that are in the footprint but do not overlap with both
- images are not copied.
- */
-template <typename ImageOrMaskedImageT>
-void copyWithinFootprint(Footprint const& foot,
-                         PTR(ImageOrMaskedImageT) const input,
-                         PTR(ImageOrMaskedImageT) output);
-
-/************************************************************************************************************/
-/**
- * \brief OR bitmask into all the Mask's pixels which are in the set of Footprint%s
- *
- * \return bitmask
- */
-template<typename MaskT>
-MaskT setMaskFromFootprintList(lsst::afw::image::Mask<MaskT> *mask,
-                               std::vector<PTR(Footprint)> const& footprints,
-                               MaskT const bitmask);
```

Comment:
Like `setImageFromFootprintList`, maybe we should reimplement this in Python so it can take an arbitrary iterable.